packages/eddsa-ticket-pcd/test/EdDSATicketPCD.spec.ts

```diff
@@ -63,6 +63,64 @@ describe("EdDSA ticket should work", function () {
     expect(await EdDSATicketPCDPackage.verify(ticket)).to.be.true;
   });
 
+  it("should not be possible to verify a ticket that has been tampered with", async function () {
+    const originalTicketData = ticket.claim.ticket;
```

Comment:
I don't think this is actually making a copy, so I think originalTicketData is being modified persistently across all of the cases.  I may be misunderstanding the Javascript, but I think you need something like:
    const originalTicketData = {...ticket.claim.ticket};

Reply:
for educational purposes:

```ts
{...ticket.claim.ticket}
```

this is good enough if `ticket.claim.ticket` is a 'plain old object', but if it contains other objects, the safer thing would be something like `JSON.parse(JSON.stringify(ticket.claim.ticket))` or the grown-up version `_.cloneDeep(ticket.claim.ticket)` (given lodash is available)

Reply:
To walk through in a bit more detail:

`ticket.claim.ticket` is is an object that is member of `ticket.claim`. `originalTicketData` is also a reference to that object.

```
ticket.claim.ticket = {
  ...originalTicketData,
  attendeeEmail: "hacker@example.com"
};
```
This creates a new object, which uses the `...` spread operator to copy the field values from the object referred to by `originalTicketData`, also setting the `attendeeEmail` field with a new value. We also set `ticket.claim.ticket` to point to this new object. So now `originalTicketData` continues to hold a reference to the object that was referred to by `ticket.claim.ticket` in the beginning, and `ticket.claim.ticket` now refers to the new object that was created using the spread operator.

Reply:
Ah, you're right.  I was expecting the copy to happen on this line, but it's happening at each of the "override" lines below instead.  I think I was confusing different types of assignment semantics in my head, sorry.

Reply:
JavaScript has multiple ways of doing the same thing, and while the `...` spread operator has the benefit of being concise it is also quite unlike anything I've seen in other languages.